org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/EclipseBuildSupport.java

```diff
@@ -49,4 +57,28 @@ public boolean isBuildFile(IResource resource) {
 		return false;
 	}
 
+	@Override
+	public boolean fileChanged(IResource resource, CHANGE_TYPE changeType, IProgressMonitor monitor) throws CoreException {
+		if (resource == null || !applies(resource.getProject())) {
+			return false;
+		}
+		refresh(resource, changeType, monitor);
+		IProject project = resource.getProject();
+		IJavaProject javaProject = JavaCore.create(project);
```

Comment:
you need to check it's a Java project beforehand

Reply:
I have updated the PR.